src/environment.rs

```diff
@@ -818,6 +818,11 @@ pub async fn update_prefix_conda(
         });
 
     let mut progress_reporter = None;
+    let mut git_reporter = None;
+    let nr_git = source_records
+        .iter()
+        .filter(|s| s.source.as_git().is_some())
+        .count();
```

Comment:
This feels a bit like a book keeping that doesnt quite belong here. Instead of having a GitReporter. Would it be possible to have a `SourceCheckoutReporter` which you construct with the initial source records that are going to be checked out if that is even needed? I can also imagine that that reporter just spawns a new progress bar when it encounters a new source checkout.